R/case_weights.R

```diff
@@ -0,0 +1,202 @@
+#' Helpers for steps with case weights
+#'
+#' These functions can be used to do basic calculations with or without case
+#' weights.
+#'
+#' @param selection A quosure
+#' @param info A data frame from the `info` argument within steps
+#' @param .data The training data
+#' @param x A numeric vector or a data frame
+#' @param wts A vector of case weights
+#' @param na_rm A logical value indicating whether `NA`
+#'  values should be removed during computations.
+#' @param use Used by [correlations()] to pass argument to [cor()]
+#' @param method Used by [correlations()] to pass argument to [cor()]
+#' @details
+#' [get_case_weights()] is designed for developers of recipe steps, to return
+#' a column with the role of "case weight" as a vector.
+#'
+#' For the other functions, rows with missing case weights are removed from
+#' calculations.
+#'
+#' For `averages()` and `variances()`, missing values in the data (*not* the
+#' case weights) only affect the calculations for those rows. For
+#' `correlations()`, the correlation matrix computation first removes rows
+#' with any missing values (equal to the "complete.obs" strategy in
+#' [stats::cor()]).
+#'
+#' `is_unsupervised_weights()`
+#' @export
+#' @name case-weight-helpers
+get_case_weights <- function(selection, info, .data) {
```

Comment:
This can be simplified. The recipe already checks the type and number of case weights. Since the weights will not be arguments to the steps, there is not need of quosures or tidyselect.